plugins/cr8-copy-s3/src/main/java/io/crate/copy/s3/S3FileInput.java

```diff
@@ -58,15 +59,15 @@ public class S3FileInput implements FileInput {
     @Nullable
     private final S3URI preGlobUri;
     @NotNull
-    private final Predicate<S3URI> uriPredicate;
+    private final Predicate<String> uriPredicate;
     @Nullable
     private final String protocolSetting;
 
     public S3FileInput(URI uri, String protocol) {
         this.clientBuilder = new S3ClientHelper();
         this.normalizedS3URI = S3URI.toS3URI(uri);
         this.preGlobUri = toPreGlobUri(this.normalizedS3URI);
-        this.uriPredicate = new GlobPredicate(this.normalizedS3URI);
+        this.uriPredicate = new GlobPredicate(this.normalizedS3URI.toString());
```

Comment:
@jeeminso, on a side note, not related to this PR:

`toS3URI` internally creates `normalizedURI` but `normalizedS3URI.toString()` returns original `URI` as a String, not normalized one. 

It means that glob predicate does pattern matching of "listed entries" against original URI. 

Original URI is provided by a user. Does it mean we have a bug in case user provided wildcard path doesn't match format of the "listing API"?

Reply:
> normalizedS3URI.toString() returns original URI as a String

`toS3URI` returns `S3URI` instance created by `return new S3URI(normalizedURI, accessKey, secretKey, bucket, key, endpoint);`. So `normalizedS3URI.toString()` should return normalized `URI`.